Mathlib/Data/Nat/Init.lean

```diff
@@ -434,6 +434,44 @@ protected lemma div_left_inj (hda : d ∣ a) (hdb : d ∣ b) : a / d = b / d ↔
   refine ⟨fun h ↦ ?_, congrArg fun b ↦ b / d⟩
   rw [← Nat.mul_div_cancel' hda, ← Nat.mul_div_cancel' hdb, h]
 
+/-- TODO: Replace `Nat.sub_mul_div` in core. -/
+lemma sub_mul_div' : (a - b * c) / b = a / b - c  := by
+  by_cases h : b * c ≤ a
+  · rw [Nat.sub_mul_div _ _ _ h]
+  · simp only [Nat.not_le] at h
+    rw [ Nat.sub_eq_zero_of_le (Nat.le_of_lt h),Nat.zero_div]
```

Comment:
If you do the correct casesplit, then there's no need to fiddle around with the hypotheses
```suggestion
  obtain h | h := le_total (b * c) a
  · rw [Nat.sub_mul_div _ _ _ h]
  · rw [ Nat.sub_eq_zero_of_le h, Nat.zero_div]
```

Reply:
Thanks. Done.